crates/wasm-pkg-common/src/registry.rs

```diff
@@ -151,6 +168,32 @@ impl RegistryMetadata {
                 .map_err(Error::InvalidRegistryMetadata)?,
         ))
     }
+
+    /// Creates a new [`RegistryMetadata`] by attempting to fetch it from a well known path. Only
+    /// errors if an error occurred with the request itself. If the URL was not found, `None` is
+    /// returned
+    pub async fn fetch(domain: &str) -> Result<Option<Self>, Error> {
```

Comment:
I'm not opposed to having this as a convenience but `reqwest` is a relatively heavyweight dependency so I'd like to gate it behind a non-default feature e.g. `metadata-client` or something.

Reply:
Yeah, that makes sense to me. I wondered about some of the heavy deps as well